packages/gatsby/src/utils/cache.js

```diff
@@ -40,13 +40,25 @@ class Cache {
 
   get(key) {
     return new Promise(resolve => {
-      this.cache.get(key, (_, res) => resolve(res))
+      this.cache.get(key, (err, res) => {
+        if (err) {
+          throw new Error(
+            `The value cached with key: ${key} could not be retrieved.`
```

Comment:
will this throw if value is not yet cached? if so this would be change in behaviour

Reply:
it's a breaking change indeed, if it's not in the cache we resolve to null or undefined probably.

Reply:
On another note, `set` would have the same issue where it would now have to be caught. Do we want to handle that differently than the current `throw` solution. 

(I apologize if my initial pull request seemed to have no regard for breaking changes. I originally wasn't planning to handle the errors and haven't ever really had to work on in production code. I appreciate all the careful reviews.)

Reply:
Don't worry @mgamsjager - I'm just trying to gather all the facts here and make sure we don't introduce breaking changes by accident ;)